src/components/Battle/Battle.jsx

```diff
@@ -0,0 +1,140 @@
+import PropTypes from 'prop-types';
+import axios from 'axios';
+import { useState, useEffect } from 'react';
+import StyleBattle from './StyleBattle';
+
+export default function Battle({ heroToFight }) {
+  const [CpuHero, setCpuHero] = useState({});
+  useEffect(() => {
+    const randomID = Math.floor(Math.random() * 730);
+    axios
+      .get(`https://superheroapi.com/api.php/4118415658211107/${randomID}`)
+      .then(({ data }) => {
+        if (
+          data.powerstats.power !== 'null' &&
+          data.powerstats.speed !== 'null' &&
+          data.powerstats.strength !== 'null' &&
+          data.image.url !== 'null'
+        ) {
+          setCpuHero(data);
+        }
+      });
+  }, []);
+
+  const fight = () => {
+    const Hero1 = {
+      strength: parseInt(heroToFight.powerstats.strength, 10),
+      speed: parseInt(heroToFight.powerstats.speed, 10),
+      durability: parseInt(heroToFight.powerstats.durability, 10),
+    };
+
+    const Hero2 = {
+      strength: parseInt(CpuHero.powerstats.strength, 10),
+      speed: parseInt(CpuHero.powerstats.speed, 10),
+      durability: parseInt(CpuHero.powerstats.durability, 10),
+    };
+    do {
+      if (
+        Hero1.speed > Hero2.speed &&
+        Hero1.durability > 0 &&
+        Hero2.durability > 0
+      ) {
+        Hero2.durability -= Hero1.strength;
+      } else if (
+        Hero2.speed > Hero1.speed &&
+        Hero1.durability > 0 &&
+        Hero2.durability > 0
+      ) {
+        Hero1.durability -= Hero2.strength;
+      } else if (
+        Hero1.speed === Hero2.speed &&
+        Hero1.durability > 0 &&
+        Hero2.durability > 0
+      ) {
+        Hero2.durability -= Hero1.strength;
+      }
+
+      if (
+        Hero1.speed > Hero2.speed &&
+        Hero1.durability > 0 &&
+        Hero2.durability > 0
+      ) {
+        Hero1.durability -= Hero2.strength;
+      } else if (
+        Hero2.speed > Hero1.speed &&
+        Hero1.durability > 0 &&
+        Hero2.durability > 0
+      ) {
+        Hero2.durability -= Hero1.strength;
+      } else if (
+        Hero1.speed === Hero2.speed &&
+        Hero1.durability > 0 &&
+        Hero2.durability > 0
+      ) {
+        Hero1.durability -= Hero2.strength;
+      }
+    } while (Hero1.durability > 0 && Hero2.durability > 0);
+
+    if (Hero1.durability > Hero2.durability) {
+      console.log('You win');
+    } else {
+      console.log('You loose');
+    }
```

Comment:
- No `console.log` allowed
- This function should return something (either `true/false` to indicate a win/loss, or the winning hero)